flake: TestUnsafeGapFillAfterSafeReorg

**Claude:** Reporting flaky test.

## Flaky Test Report

**Test:** `TestUnsafeGapFillAfterSafeReorg`
**Package:** `github.com/ethereum-optimism/optimism/op-acceptance-tests/tests/sync/elsync/reorg`
**Job:** `memory-all-opn-op-geth`

## Failure Details

**Error:**
```
Error Trace: /home/circleci/project/op-acceptance-tests/tests/sync/elsync/reorg/sync_test.go:84
Error: "2" is not greater than or equal to "3"
Test: TestUnsafeGapFillAfterSafeReorg
```

**Duration:** 61.29s

## CI Link

https://app.circleci.com/pipelines/github/ethereum-optimism/optimism/121550/workflows/552d84df-4c8f-4d24-a572-260daa8baa8e/jobs/4725058/tests

## Context

The test asserts that some count (likely unsafe blocks) is >= 3, but only 2 were observed. This suggests a timing-dependent condition where the system hasn't produced enough blocks by the time the assertion fires.

**Claude:** Reopening — test is still flaking.

CircleCI Insights shows **2 recent flakes** on `memory-all-opn-op-reth`.

Latest failure on PR #20640 (`kona-executor` Rust change, unrelated to test):
- CI: https://app.circleci.com/pipelines/gh/ethereum-optimism/optimism/124730/workflows/53c5dc24-2973-49fb-a6a6-8423949183bf
- Error trace: `op-acceptance-tests/tests/sync/elsync/reorg/sync_test.go:48`
- Error: `Received unexpected error: Post "http://127.0.0.1:37577/sequencers/test-seq-900": context deadline exceeded`
- Followed by: `assertions.go:387: Error: Condition never satisfied`

Investigation underway.

--- a/op-acceptance-tests/tests/sync/elsync/reorg/sync_test.go
+++ b/op-acceptance-tests/tests/sync/elsync/reorg/sync_test.go
@@ -9,7 +9,6 @@ import (
 	"github.com/ethereum-optimism/optimism/op-service/eth"
 	"github.com/ethereum-optimism/optimism/op-supervisor/supervisor/types"
 	"github.com/ethereum-optimism/optimism/op-test-sequencer/sequencer/seqtypes"
-	"github.com/ethereum/go-ethereum/common"
 )
 
 // TestUnsafeGapFillAfterSafeReorg demonstrates the sequence:
@@ -43,9 +42,19 @@ func TestUnsafeGapFillAfterSafeReorg(gt *testing.T) {
 	startL1Block := sys.L1EL.BlockRefByLabel(eth.Unsafe)
 
 	require.Eventually(func() bool {
-		// Advance single L1 block
-		require.NoError(ts.New(ctx, seqtypes.BuildOpts{Parent: common.Hash{}}))
-		require.NoError(ts.Next(ctx))
+		// Advance a single L1 block. Sequencer.Next internally calls New with
+		// empty BuildOpts and tolerates ErrConflictingJob, so we do not call
+		// ts.New here — that would fail with ErrConflictingJob if a previous
+		// Next attempt timed out and left the job state wedged.
+		//
+		// We must not use require.NoError inside this polling callback: a
+		// single transient engine-API stall (CPU starvation under CI load)
+		// would otherwise mark the test failed on the first error. Instead we
+		// log and return false so Eventually retries until the L1 EL recovers.
+		if err := ts.Next(ctx); err != nil {
+			logger.Warn("ts.Next failed, will retry", "err", err)
+			return false
+		}
 		l1head := sys.L1EL.BlockRefByLabel(eth.Unsafe)
 		l2Safe := sys.L2EL.BlockRefByLabel(eth.Safe)
 		logger.Info("l1 info", "l1_head", l1head, "l1_origin", l2Safe.L1Origin, "l2Safe", l2Safe)
@@ -141,9 +150,13 @@ func TestUnsafeGapFillAfterUnsafeReorg_RestartL2CL(gt *testing.T) {
 	startL1Block := sys.L1EL.BlockRefByLabel(eth.Unsafe)
 
 	require.Eventually(func() bool {
-		// Advance single L1 block
-		require.NoError(ts.New(ctx, seqtypes.BuildOpts{Parent: common.Hash{}}))
-		require.NoError(ts.Next(ctx))
+		// Advance a single L1 block. See the comment on the matching loop in
+		// TestUnsafeGapFillAfterSafeReorg for why we use ts.Next alone and do
+		// not require.NoError inside the polling callback.
+		if err := ts.Next(ctx); err != nil {
+			logger.Warn("ts.Next failed, will retry", "err", err)
+			return false
+		}
 		l1head := sys.L1EL.BlockRefByLabel(eth.Unsafe)
 		l2Unsafe := sys.L2EL.BlockRefByLabel(eth.Unsafe)
 		logger.Info("l1 info", "l1_head", l1head, "l1_origin", l2Unsafe.L1Origin, "l2Unsafe", l2Unsafe)
@@ -281,9 +294,13 @@ func TestUnsafeGapFillAfterUnsafeReorg_RestartCLP2P(gt *testing.T) {
 	startL1Block := sys.L1EL.BlockRefByLabel(eth.Unsafe)
 
 	require.Eventually(func() bool {
-		// Advance single L1 block
-		require.NoError(ts.New(ctx, seqtypes.BuildOpts{Parent: common.Hash{}}))
-		require.NoError(ts.Next(ctx))
+		// Advance a single L1 block. See the comment on the matching loop in
+		// TestUnsafeGapFillAfterSafeReorg for why we use ts.Next alone and do
+		// not require.NoError inside the polling callback.
+		if err := ts.Next(ctx); err != nil {
+			logger.Warn("ts.Next failed, will retry", "err", err)
+			return false
+		}
 		l1head := sys.L1EL.BlockRefByLabel(eth.Unsafe)
 		l2Unsafe := sys.L2EL.BlockRefByLabel(eth.Unsafe)
 		logger.Info("l1 info", "l1_head", l1head, "l1_origin", l2Unsafe.L1Origin, "l2Unsafe", l2Unsafe)

--- a/op-acceptance-tests/tests/sync/follow_l2/sync_test.go
+++ b/op-acceptance-tests/tests/sync/follow_l2/sync_test.go
@@ -10,7 +10,6 @@ import (
 	"github.com/ethereum-optimism/optimism/op-service/eth"
 	"github.com/ethereum-optimism/optimism/op-supervisor/supervisor/types"
 	"github.com/ethereum-optimism/optimism/op-test-sequencer/sequencer/seqtypes"
-	"github.com/ethereum/go-ethereum/common"
 )
 
 func TestFollowL2_Safe_Finalized_CurrentL1(gt *testing.T) {
@@ -99,9 +98,19 @@ func TestFollowL2_ReorgRecovery(gt *testing.T) {
 	startL1Block := sys.L1EL.BlockRefByLabel(eth.Unsafe)
 
 	require.Eventually(func() bool {
-		// Advance single L1 block
-		require.NoError(ts.New(ctx, seqtypes.BuildOpts{Parent: common.Hash{}}))
-		require.NoError(ts.Next(ctx))
+		// Advance a single L1 block. Sequencer.Next internally calls New with
+		// empty BuildOpts and tolerates ErrConflictingJob, so we do not call
+		// ts.New here — that would fail with ErrConflictingJob if a previous
+		// Next attempt timed out and left the job state wedged.
+		//
+		// We must not use require.NoError inside this polling callback: a
+		// single transient engine-API stall (CPU starvation under CI load)
+		// would otherwise mark the test failed on the first error. Instead we
+		// log and return false so Eventually retries until the L1 EL recovers.
+		if err := ts.Next(ctx); err != nil {
+			logger.Warn("ts.Next failed, will retry", "err", err)
+			return false
+		}
 		l1head := sys.L1EL.BlockRefByLabel(eth.Unsafe)
 		l2Safe := sys.L2ELB.BlockRefByLabel(eth.Safe)
 

